internal/extrafiles/extra_files.go

```diff
@@ -6,31 +6,35 @@ import (
 	"path/filepath"
 
 	"github.com/apex/log"
+	"github.com/goreleaser/fileglob"
 	"github.com/goreleaser/goreleaser/pkg/config"
-	"github.com/mattn/go-zglob"
 )
 
 // Find resolves extra files globs et al into a map of names/paths or an error.
 func Find(files []config.ExtraFile) (map[string]string, error) {
 	var result = map[string]string{}
 	for _, extra := range files {
-		if extra.Glob != "" {
-			files, err := zglob.Glob(extra.Glob)
-			if err != nil {
-				return result, fmt.Errorf("globbing failed for pattern %s: %w", extra.Glob, err)
+		if extra.Glob == "" {
+			continue
+		}
+		files, err := fileglob.Glob(extra.Glob)
+		if err != nil {
+			return result, fmt.Errorf("globbing failed for pattern %s: %w", extra.Glob, err)
+		}
+		if len(files) == 0 {
+			return result, fmt.Errorf("globbing failed for pattern %s: file does not exist", extra.Glob)
```

Comment:
zglob handled it differently, actually erroring if file not found...

Reply:
The `file does not exist` seems a bit weird. The actual message should depend on the syntax tree of the glob expression:
* Glob only statically matches a single file which does not exist -> `file does not exist`
* Glob contains wildcard (or similar matchers) -> `no matching files`

We could do this in `fileglob.Glob` by checking if the static prefix is identical to the glob expression.

Reply:
hmm, yes, that makes sense, will add it there.